pkg/doc/did/doc.go

```diff
@@ -0,0 +1,207 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package did
+
+import (
+	"encoding/json"
+	"encoding/pem"
+
+	"encoding/hex"
+
+	"github.com/btcsuite/btcutil/base58"
+	"github.com/pkg/errors"
+)
+
+const (
+	jsonldType         = "type"
+	jsonldID           = "id"
+	jsonldServicePoint = "serviceEndpoint"
+	jsonldController   = "controller"
+
+	// various public key encodings
+	jsonldPublicKeyBase58 = "publicKeyBase58"
+	jsonldPublicKeyHex    = "publicKeyHex"
+	jsonldPublicKeyPem    = "publicKeyPem"
+)
+
+// Doc DID Document definition
+type Doc struct {
+	Context        []string
+	ID             string
+	PublicKey      []PublicKey
+	Service        []Service
+	Authentication []VerificationMethod
+	Created        string
```

Comment:
@fqutishat `Created` and `Updated` should be `time.Time`

Reply:
done